Another clean-up

nima opened the pull request:
A bug was introduced in the last cleanup branch. There were no tests for this code path, so it was never caught.

baua reviewed (approved):
Easy enough.

nima committed:
Fix suffixed function test issue

nima committed:
Be smart about user-supplied `--'

baua commented on lib/libsh/libsimbol/libsimbol.sh:
```diff
@@ -1516,6 +1505,7 @@ EXCEPTION_DEPRECATED=72
 EXCEPTION_MISSING_PERL_MOD=80
 EXCEPTION_MISSING_PYTHON_MOD=81
 EXCEPTION_MISSING_USER=82
+EXCEPTION_USER_ERROR=82
```
Seems to be the same as EXCEPTION_MISSING_USER

baua commented on lib/libsh/libsimbol/libsimbol.sh:
```diff
@@ -1572,4 +1563,86 @@ function core:raise() {
+
+        core:bashenv append <<MOCKERY
+function ${mocked}() {
+    $(for statement in "${statements[@]}"; do echo "    ${statement};"; done)
```
Where is statements coming from?

baua replied:
Ah shit. Got it. I got mocked.

nima commented:
Oops!    Good catch! Ill fix it and push again!    Next up is the other branch on standardization of tests (and adding more of)... If ever to 80% code coverage we will be close to shipping v1!

nima committed:
Set `travis' to `fast-finish'

Some of the failing tests take as long as Travis gives them, and
that's too long.

General cleanup.

nima commented:
I took out all the features that crept into this branch, keeping this simple so we can close early and test each feature in isolation on top of the new `develop` branch once this is closed and merged.

nima committed:
Exec to SIMBOL_SHELL, not merely call it

nima committed:
Fix missed rename on a USER variable
